rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/internal/LRARecoveryModule.java

```diff
@@ -98,7 +99,11 @@ private void doRecoverTransaction(Uid recoverUid) {
             boolean inFlight = (lra.getLRAStatus() == LRAStatus.Active);
 
             LRAStatus lraStatus = lra.getLRAStatus();
-            if (LRAStatus.FailedToCancel.equals(lraStatus) || LRAStatus.FailedToClose.equals(lraStatus)) {
+
+            // This condition is to check if there in any failed LRA record
+            if (lra.isFailed()) {
+                // Moving this failed record to another part of store
+                moveEntryToFailedLRAPath(recoverUid);
```

Comment:
Here also it is the same scenario as I mentioned above. If there is a failed record present then, it would not be moved to failed location until recoveryManager executes its next run, so before that we need to check both the locations.